llpc/util/llpcElfWriter.cpp

```diff
@@ -479,10 +479,14 @@ template <class Elf> void ElfWriter<Elf>::assembleNotes() {
     memcpy(data, &note.hdr, noteHeaderSize);
     data += noteHeaderSize;
     const unsigned noteNameSize = alignTo(note.hdr.nameSize, sizeof(unsigned));
-    memcpy(data, &note.hdr.name, noteNameSize);
+    memcpy(data, &note.hdr.name, note.hdr.nameSize);
+    for (unsigned i = note.hdr.nameSize; i < noteNameSize; i++)
+      data[i] = 0;
```

Comment:
If the memcpy copies 'too little' data now, wouldn't this be solved by zero-initializing the buffer when you allocate it?

Reply:
Yes, that would have the same effect. I thought zeroing only the needed part is little more efficient.

Reply:
I think memset is more clear. If I remember correctly, PAL team once discussed this when drafting the coding style and recommended the use of memset for initialization because of its high efficiency.